File: src/com/nerdscorner/android/plugin/github/ui/tablemodels/GHPullRequestTableModel.java
package com.nerdscorner.android.plugin.github.ui.tablemodels;

import org.kohsuke.github.GHPullRequest;

import java.io.IOException;
import java.io.Serializable;
import java.text.SimpleDateFormat;
import java.util.List;

import com.nerdscorner.android.plugin.github.domain.gh.GHPullRequestWrapper;
import com.nerdscorner.android.plugin.utils.Strings;

public class GHPullRequestTableModel extends BaseModel<GHPullRequestWrapper> implements Serializable {
    private static final int COLUMN_TITLE = 0;
    private static final int COLUMN_AUTHOR = 1;
    private static final int COLUMN_DATE = 2;

    public GHPullRequestTableModel(List<GHPullRequestWrapper> pullRequests, String[] colNames) {
        super(pullRequests, colNames);
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        if (rowIndex < 0 || rowIndex >= items.size()) {
            return null;
        }
        GHPullRequestWrapper pullRequestWrapper = items.get(rowIndex);
        GHPullRequest pullRequest = pullRequestWrapper.getGhPullRequest();
        switch (columnIndex) {
            case COLUMN_TITLE:
                return pullRequest.getTitle();
            case COLUMN_AUTHOR:
                try {
                    return pullRequest.getUser().getLogin();
                } catch (Exception e) {
                    return null;
                }
            case COLUMN_DATE:
                try {
                    return new SimpleDateFormat(Strings.DATE_FORMAT).format(pullRequest.getUpdatedAt());
                } catch (Exception e) {
                    e.printStackTrace();
                }
        }
        return null;
    }
}

File: src/com/nerdscorner/android/plugin/github/ui/tablemodels/GHPullRequestTableModel.kt
package com.nerdscorner.android.plugin.github.ui.tablemodels

import com.nerdscorner.android.plugin.github.domain.gh.GHPullRequestWrapper
import com.nerdscorner.android.plugin.utils.Strings
import java.io.Serializable
import java.text.SimpleDateFormat

class GHPullRequestTableModel(pullRequests: MutableList<GHPullRequestWrapper>, colNames: Array<String>) :
        BaseModel<GHPullRequestWrapper>(pullRequests, colNames), Serializable {

    override fun getValueAt(rowIndex: Int, columnIndex: Int): Any? {
        if (rowIndex < 0 || rowIndex >= items.size) {
            return null
        }
        val pullRequestWrapper = items[rowIndex]
        val pullRequest = pullRequestWrapper.ghPullRequest
        return when (columnIndex) {
            COLUMN_TITLE -> return pullRequest.title
            COLUMN_AUTHOR -> {
                try {
                    pullRequest.user.login
                } catch (e: Exception) {
                    null
                }
            }
            COLUMN_DATE -> try {
                SimpleDateFormat(Strings.DATE_FORMAT).format(pullRequest.updatedAt)
            } catch (e: Exception) {
                e.printStackTrace()
            }
            COLUMN_CI_URL -> {
                pullRequestWrapper.buildStatus
            }
            else -> null
        }
    }

    companion object {
        private const val COLUMN_TITLE = 0
        private const val COLUMN_AUTHOR = 1
        private const val COLUMN_DATE = 2
        const val COLUMN_CI_URL = 3
    }
}
